### PR TITLE
Cache num finished mentoring sessions

### PR DESCRIPTION
**Pull request by ErikSchierboom**
- Add num_finished_mentoring_sessions to user_track_mentorships
- Convert mentor stats updating from job to regular command
- Cache num finished discussions
- Cache num finished discussions

### CODE REVIEW

File: app/commands/mentor/update_stats.rb [hunk 1]
```diff
@@ -1,7 +1,11 @@
-class UpdateMentorStatsJob < ApplicationJob
+class Mentor::UpdateStats
```
Comment (ErikSchierboom) [inline app/commands/mentor/update_stats.rb:1]: As this wasn't used in our sidekiq schedule, I snuck in this additional refactoring to make this a mandate job

Comment (iHiD) [reply app/commands/mentor/update_stats.rb:1]: There's no syntax highlighting in this file for some reason. Double-check the file extension doesn't have any hidden unicode pls.

File: app/models/mentor/discussion.rb [hunk 2]
```diff
@@ -159,10 +159,11 @@ def awaiting_mentor!
   end
 
   def update_stats!
-    UpdateMentorStatsJob.perform_later(
+    Mentor::UpdateStats.defer(
       mentor,
       update_num_solutions_mentored: previous_changes.key?('status'),
       update_satisfaction_rating: previous_changes.key?('rating')
     )
+    Mentor::Discussion::UpdateNumFinishedDiscussions.defer(self) if previous_changes.key?('status') && finished?
```
Comment (iHiD) [inline app/models/mentor/discussion.rb:167]: Nice.

File: db/migrate/20221201144756_add_num_finished_discussions_to_user_track_mentorships.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+class AddNumFinishedDiscussionsToUserTrackMentorships < ActiveRecord::Migration[7.0]
+  def change
+    unless Rails.env.production?
+      add_column :user_track_mentorships, :num_finished_discussions, :integer, null: false, default: 0
```
Comment (iHiD) [inline db/migrate/20221201144756_add_num_finished_discussions_to_user_track_mentorships.rb]: mediumint?

**Review by iHiD — APPROVED** [review]

**Pull request merged**